orchagent/muxorch.cpp

```diff
@@ -983,6 +1075,91 @@ sai_object_id_t MuxOrch::getNextHopTunnelId(std::string tunnelKey, IpAddress& ip
     return it->second.nh_id;
 }
 
+/**
+ * @brief updates the given route to point to a single active NH or tunnel
+ * @param pfx IpPrefix of route to update
+ * @param remove bool only true when route is getting removed
+ */
+void MuxOrch::updateRoute(const IpPrefix &pfx, bool remove)
+{
+    NextHopGroupKey nhg_key;
+    NextHopGroupEntry nhg_entry;
+
+    if (remove)
+    {
+        mux_multi_nh_route_tb.erase(pfx);
+        return;
+    }
+
+    /* get nexthop group key from syncd */
+    nhg_key = gRouteOrch->getSyncdRouteNhgKey(gVirtualRouterId, pfx);
+
+    /* check for cross-mux neighbors */
+    if (nhg_key.getSize() > 1)
+    {
+        std::set<NextHopKey> nextHops;
+        sai_object_id_t next_hop_id;
+        sai_status_t status;
+        bool active_found = false;
+
+        /* get nexthops from nexthop group */
+        nextHops = nhg_key.getNextHops();
+
+        auto it = mux_multi_nh_route_tb.find(pfx);
+        if (it != mux_multi_nh_route_tb.end())
+        {
+            MuxCable* cable = findMuxCableInSubnet(it->second.ip_address);
+            if (cable == nullptr || cable->isActive())
```

Comment:
If I'm reading this correctly, if our route nexthop is not one of the configured mux cable IPs, we take no action. This seems a bit optimistic to me, unless we can 100% guarantee that the multi-mux-nexthop scenario only occurs with nexthop IPs that are configured as mux cable IPs?

Reply:
From my understanding from discussions with Prince we currently handle neighbors on the Vlan that aren't tied to a muxcable as active. maybe we can have a discussion on this?

Reply:
i think `findMuxCableInSubnet` returns if its directly connected server IP but not all learned neighbors (and is in config_db). But i think we have to handle it if the nexthop belongs to any mux cable.